### PR TITLE
Fixes headslug interaction with mindswapping

### PR DESCRIPTION
**Pull request by ChesterTheCheesy**
# General Documentation
### Intent of your Pull Request
For some god unknown reasons, headslug stores the mind of original owner of the body, and passes it down when laying eggs.
This means, if you had a changeling that turned into a headslug, and then someone mindswapped with the headslug, resulting egg hatch results in the old mind being put in there. This clearly isn't intended.

### Why is this change good for the game?
Allows people to mindswap into changeling headslugs properly.

### Briefly describe your PR and the impacts of it, in layman's terms. 
makes the headslug pass its current mind into the egg, instead of a variable that isn't changed when mindswapped

### What general grouping does this PR fall under? 
extremely minor bugfix to headslug

# Changelog
:cl:  
bugfix: headslugs now properly pass their current mind onto eggs
/:cl:

### CODE REVIEW

**Comment by ChesterTheCheesy** [comment]
Closing because people still don't like xenolings, even stolen from a real ling.

**Comment by MegaEmpirical** [comment]
See you saying a wizard couldn't swap with a head slug and successfully implant? Currently.
